internal/datastore/ydb/common/common.go

```diff
@@ -0,0 +1,57 @@
+package common
+
+import (
+	"bytes"
+	"net/url"
+	"sync"
+)
+
+type DSN struct {
+	OriginalDSN     string
+	TablePathPrefix string
+}
+
+// ParseDSN is used to extract custom parameters from YDB DSN that are used to alter datastore behaviour.
+// The following parameters are recognized:
+//
+//   - table_path_prefix: string. Will be added to all queries.
+func ParseDSN(dsn string) DSN {
+	parsedDSN := DSN{OriginalDSN: dsn}
+
+	uri, err := url.Parse(dsn)
+	if err != nil {
+		// don't care about the error since ydb.Open will parse dsn again.
+		return parsedDSN
+	}
+
+	params := uri.Query()
+	parsedDSN.TablePathPrefix = params.Get("table_path_prefix")
```

Comment:
If the parameter is not specified or is empty, is this an error?

Reply:
From Get docs:
>  If there are no values associated with the key, Get returns the empty string

So effectively no table prefix will be ever added. All SpiceDB tables will be places in YDB database's root directory&